.circleci/config.yml

```diff
@@ -10,24 +30,82 @@ jobs:
     steps:
       - checkout
       - restore_caches
-      - setup_rust
-      - setup_system
+      - install_rust
+      - install_node_devlibs
       - print_current_versions
-
       - run:
-          name: Build using cargo make
+          name: Debug build and test using cargo make
           command: cargo make ci
+      - store_artifacts:
+          path: api_tests/log
+      - save_caches
 
+  release-build-test:
+    working_directory: ~/comit
+    machine:
+      image: ubuntu-1604:201903-01
+      docker_layer_caching: true
+    environment:
+      RUST_TEST_THREADS: "8"
+    steps:
+      - checkout
+      - restore_caches
+      - install_rust
+      - install_node_devlibs
+      - print_current_versions
+      - run:
+          name: Release build and test using cargo make
+          command: |
+            export CARGO_MAKE_CARGO_BUILD_TEST_FLAGS="--all --release"
+            export BTSIEVE_BIN=~/comit/target/release/btsieve
+            export CND_BIN=~/comit/target/release/cnd
+            cargo make ci
       - store_artifacts:
           path: api_tests/log
+      - run:
+          name: Consolidate release binaries
+          command: |
+            mkdir ~/artifacts
+            mv -v ~/comit/target/release/cnd ~/artifacts
+            mv -v ~/comit/target/release/btsieve ~/artifacts
+      - persist_to_workspace:
+          root: ~/artifacts
+          paths:
+            - cnd
+            - btsieve
 
-      - save_caches
+  publish-github-release:
+    working_directory: ~/comit
+    machine:
+      image: ubuntu-1604:201903-01
+      docker_layer_caching: true
+    environment:
+      RUST_TEST_THREADS: "8"
+    steps:
+      - attach_workspace:
+          at: ~/artifacts
+      - checkout
+      - restore_caches
+      - install_rust
+      - install_node_devlibs
+      - print_current_versions
+      - run:
+          name: "Package & Publish Release on GitHub"
+          command: |
+            go get github.com/tcnksm/ghr
+            VERSION=$(cargo pkgid -- cnd|cut -d# -f2)
+
+            # Let's abort if the git tag does not match the version in Cargo.toml as it would be fishy
+            test "${VERSION}" = "${CIRCLE_TAG}"
+            mkdir ~/package
+            cd ~/artifacts && tar czvf ~/package/comit-rs_v${VERSION}_$(uname -s)_$(uname -m).tar.gz *
+            ghr -t ${GITHUB_TOKEN_FOR_RELEASES} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME} -c ${CIRCLE_SHA1} -delete ${VERSION} ~/package/
 
 commands:
-  setup_system:
+  install_node_devlibs:
```

Comment:
What was wrong about `setup_system`?
Your current name doesn't capture that we kill `systemd` for example ..

Reply:
Setup system does not really gives you much about what it actually does. Also, the systemd is a side effect of achievement the goal of the command: installing node and dev libs

Reply:
Can we do `install_node_and_devlibs` then? Currently it reads like we are installing node development libraries.

Reply:
Please do a follow-up PR please, I am want to merge this :)